instrumentation-docs/src/test/java/io/opentelemetry/instrumentation/docs/ManualTelemetryTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.docs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.opentelemetry.instrumentation.docs.internal.InstrumentationMetadata;
+import io.opentelemetry.instrumentation.docs.internal.ManualTelemetryEntry;
+import io.opentelemetry.instrumentation.docs.utils.YamlHelper;
+import org.junit.jupiter.api.Test;
+
+class ManualTelemetryTest {
+
+  @Test
+  void testManualTelemetryParsing() throws JsonProcessingException {
+    String yamlContent =
+        """
+        description: "Example instrumentation with manual telemetry documentation"
+        semantic_conventions:
+          - HTTP_CLIENT_SPANS
+        library_link: https://example.com/library
+        additional_telemetry:
+        - when: default
+          metrics:
+          - name: system.disk.io
+            description: System disk IO
+            type: LONG_SUM
+            unit: By
+            attributes:
+            - name: device
+              type: STRING
+            - name: direction
+              type: STRING
+          spans:
+          - span_kind: CLIENT
+            attributes:
+            - name: custom.operation
+              type: STRING
+        - when: experimental
+          metrics:
+          - name: experimental.feature.usage
+            description: Usage of experimental features
+            type: HISTOGRAM
+            unit: s
+            attributes:
+            - name: feature.name
+              type: STRING
+        """;
+
+    InstrumentationMetadata metadata = YamlHelper.metaDataParser(yamlContent);
+
+    assertThat(metadata).isNotNull();
```

Comment:
rather than these long chains of asserts, you could build the expected object using the builder and then compare that against the parsed with `.usingRecursiveComparison()` - this is done in the SDK, e.g. https://github.com/open-telemetry/opentelemetry-java/blob/2c0ee00508d4c7ecc8f9d2306e0e5948a5f28660/sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/DeclarativeConfigurationParseTest.java#L767